subsys/usb/device_next/class/usbd_cdc_ncm.c

```diff
@@ -0,0 +1,1278 @@
+	}
+	while (sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex) != 0
+			&&  sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramLength) != 0) {
+		LOG_DBG("  << %d %d", sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex),
+				sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramLength));
+		if (sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex) > len) {
+			LOG_ERR("(EE) ill start of datagram[%d]: %d (%d)",
+					ndx,
+					sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex), len);
+			return false;
+		}
+		if (sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex)
+				+ sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramLength) > len) {
+			LOG_ERR("(EE) ill end of datagram[%d]: %d (%d)", ndx,
+				sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramIndex)
+				+ sys_le16_to_cpu(ndp16_datagram[ndx].wDatagramLength), len);
+			return false;
+		}
+		++ndx;
+	}
+
+	LOG_HEXDUMP_DBG(ntb->data, len, "NTB");
+
+	/* -> ntb contains a valid packet structure */
+	return true;
+}   /* _cdc_ncm_frame_ok */
+
+
+
+static int cdc_ncm_acl_out_cb(struct usbd_class_data *const c_data,
+		struct net_buf *const buf, const int err)
+/**
+ * Frame received from host!
+ */
+{
+	const struct device *dev = usbd_class_get_private(c_data);
+	struct cdc_ncm_eth_data *data = dev->data;
+	struct net_pkt *pkt;
+	union recv_ntb_t *ntb;
+	struct ndp16_datagram_t *ndp_datagram;
+
+	LOG_DBG("len %d err %d", buf->len, err);
+
+	if (err  ||  buf->len == 0) {
+		goto restart_out_transfer;
+	}
+
+	/* LOG_HEXDUMP_DBG(buf->data, buf->len, "ntb"); */
+
+	if (!_cdc_ncm_frame_ok(data, buf)) {
+		LOG_ERR("ill frame received from host");
+		goto restart_out_transfer;
+	}
+
+	ntb = (union recv_ntb_t *)buf->data;
+	ndp_datagram = (struct ndp16_datagram_t *)
+			(ntb->data + sys_le16_to_cpu(ntb->nth.wNdpIndex) + sizeof(struct ndp16_t));
+
+	uint16_t start = sys_le16_to_cpu(ndp_datagram[0].wDatagramIndex);
+	uint16_t len   = sys_le16_to_cpu(ndp_datagram[0].wDatagramLength);
+
+	pkt = net_pkt_rx_alloc_with_buffer(data->iface, len, AF_UNSPEC, 0, K_FOREVER);
+	if (pkt == NULL) {
+		LOG_ERR("No memory for net_pkt");
+		goto restart_out_transfer;
+	}
+
+	/*    LOG_HEXDUMP_DBG(buf->data + start, len, "frame"); */
+
+	if (net_pkt_write(pkt, ntb->data + start, len)) {
+		LOG_ERR("Unable to write into pkt");
+		net_pkt_unref(pkt);
+		goto restart_out_transfer;
+	}
+
+	LOG_DBG("Received packet len %zu", len);
+	if (net_recv_data(data->iface, pkt) < 0) {
+		LOG_ERR("Packet %p dropped by network stack", pkt);
+		net_pkt_unref(pkt);
+	}
+
+restart_out_transfer:
+	net_buf_unref(buf);
+	atomic_clear_bit(&data->state, CDC_NCM_OUT_ENGAGED);
+
+	return cdc_ncm_out_start(c_data);
+}   /* cdc_ncm_acl_out_cb */
+
+
+
+static int _usbd_cdc_ncm_send_notification(const struct device *dev,
+		const void *notification, uint16_t len)
+/**
+ * Send a notification to the host.
+ */
+{
+	struct cdc_ncm_eth_data *data = dev->data;
+	struct usbd_class_data *c_data = data->c_data;
+	struct net_buf *buf;
+	uint8_t ep;
+	int ret;
+
+	LOG_DBG("");
```

Comment:
I suppose this would receive similar comment to this:
https://github.com/zephyrproject-rtos/zephyr/pull/53408#discussion_r1182371092

A `git commit --amend` adding some short description of just a few words would allow keeping these.

Reply:
removed every empty LOG_DBG(), wondering if anybody ever made the cost/environmental calculation of these CI runs caused by mini changes...

Reply:
I read somewhere that scripts/rules were setup to avoid extra CI cycles, with manual or auto-run CI.

Maybe there can be rules to only auto-run CI when the PR is not in "draft" state.